lib/assignDisabledRanges.js

```diff
@@ -66,38 +85,43 @@ module.exports = function(
 
     if (ruleName === ALL_RULES) {
       Object.keys(disabledRanges).forEach(disabledRuleName => {
-        startDisabledRange(line, disabledRuleName);
-        endDisabledRange(line, disabledRuleName);
+        const strict = disabledRuleName === ALL_RULES;
+
+        startDisabledRange(line, disabledRuleName, strict);
+        endDisabledRange(line, disabledRuleName, strict);
```

Comment:
regarding to `strict` flag definition - if `ruleName` is `all`, then `strictStart`/`strictEnd` true, otherwise false